go/mysql/flavor.go

```diff
@@ -163,6 +163,18 @@ func (c *Conn) MasterPosition() (Position, error) {
 	}, nil
 }
 
+// MasterFilePosition returns the current master's file based replication position.
+func (c *Conn) MasterFilePosition() (Position, error) {
+	filePosFlavor := filePosFlavor{}
+	gtidSet, err := filePosFlavor.masterGTIDSet(c)
+	if err != nil {
+		return Position{}, err
+	}
+	return Position{
+		GTIDSet: gtidSet,
+	}, nil
```

Comment:
Merely a coding style comment, feel free to ignore. What I usually do is:
```suggestion
	return Position{
		GTIDSet: gtidSet,
	}, err
```
it's the caller's responsibility to check if `err != nil` and if so, to ignore the `GTIDSet` value.

Reply:
I followed the pattern established in the rest of the codebase but you bring up a good point. I'm somewhat split on this. I agree with you that it's the caller's responsibility to check if `err != nil` but I'm also used to writing in languages where if a result is errorful, it's not even possible to access the successful result. 

In this case though if err is not nil, then gtidSet would be nil anyway, so I think your version would actually return the same result with less code - and I like that. I suppose it's a question of how others feel regarding established patterns that already exist in the Vitess codebase, and if we should follow those patterns or break them when it's sensible? @enisoc @deepthi @systay @sougou?

Reply:
I prefer the less compact version because it is easier to read.